internal/services/containers/kubernetes_cluster_node_pool_resource.go

```diff
@@ -397,6 +402,7 @@ func resourceKubernetesClusterNodePoolCreate(d *pluginsdk.ResourceData, meta int
 		Type:                   containerservice.AgentPoolTypeVirtualMachineScaleSets,
 		VMSize:                 utils.String(d.Get("vm_size").(string)),
 		UpgradeSettings:        expandUpgradeSettings(d.Get("upgrade_settings").([]interface{})),
+		HostGroupID:            utils.String(d.Get("host_group_id").(string)),
```

Comment:
Remove this line, because we have this block below
```
if HostGroupID := d.Get("host_group_id").(string); HostGroupID != "" {
		profile.HostGroupID = utils.String(HostGroupID)
	}
```